prow/pod-utils/decorate/podspec_test.go

```diff
@@ -0,0 +1,329 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package decorate
+
+import (
+	"testing"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/diff"
+
+	"k8s.io/test-infra/prow/kube"
+)
+
+func TestProwJobToPod(t *testing.T) {
+	tests := []struct {
+		podName string
+		buildID string
+		labels  map[string]string
+		pjSpec  kube.ProwJobSpec
+
+		expected *v1.Pod
+	}{
+		{
+			podName: "pod",
+			buildID: "blabla",
+			labels:  map[string]string{"needstobe": "inherited"},
+			pjSpec: kube.ProwJobSpec{
```

Comment:
I feel like this would be easier for humans to read as input prow yaml, output kubernetes resource yaml, rather than creating structs?

Reply:
(I personally feel strongly in the other direction)